crypto/evp/pmeth_lib.c

```diff
@@ -422,6 +430,38 @@ int EVP_PKEY_CTX_set_dh_pad(EVP_PKEY_CTX *ctx, int pad)
 }
 #endif
 
+int EVP_PKEY_CTX_set_signature_md(EVP_PKEY_CTX *ctx, const EVP_MD *md)
+{
+    OSSL_PARAM sig_md_params[3];
```

Comment:
Can you give me a reference to where you've done that? I can't seem to find one.

Reply:
You need to look in @paulidale's KDF port.  But essentially, it allows this:

``` C
     *p++ = OSSL_PARAM_construct_utf8_string(
                           OSSL_SIGNATURE_PARAM_MD_NAME,
                           /*
                            * Cast away the const. This is read only so should
                            * be safe
                            */
                           (char *)name,
                           strlen(name) + 1);
    *p++ = OSSL_PARAM_construct_size_t(OSSL_SIGNATURE_PARAM_MD_SIZE, &mdsize);
    *p++ = OSSL_PARAM_construct_end();
```

... instead of

``` C
     sig_md_params[0] = OSSL_PARAM_construct_utf8_string(
                           OSSL_SIGNATURE_PARAM_MD_NAME,
                           /*
                            * Cast away the const. This is read only so should
                            * be safe
                            */
                           (char *)name,
                           strlen(name) + 1);
    sig_md_params[1] = OSSL_PARAM_construct_size_t(OSSL_SIGNATURE_PARAM_MD_SIZE,
                                                   &mdsize);
    sig_md_params[2] = OSSL_PARAM_construct_end();
```

Reply:
I've used this construction now.